File: framework/scala/src/main/scala/soda/scala/web/MessageWriter.scala
package soda.scala.web

import com.sun.net.httpserver.HttpExchange

import java.nio.charset.StandardCharsets

trait MessageWriter {

  def sendMessage(exchange: HttpExchange, code: Int, message: String): Unit = {
    val data = message.getBytes(StandardCharsets.UTF_8)
    exchange.sendResponseHeaders(code, data.length)
    exchange.getResponseBody.write(data)
    exchange.getResponseBody.close()
  }

  def sendMessageWithCatch(exchange: HttpExchange, code: Int, message: String): Unit = {
    try {
      sendMessage(exchange, code, message)
    } catch {
      case ex: Throwable => Logger.exception(s"message sending error, code - $code, msg - $message", ex)
    }
  }

}

File: framework/kotlin/src/main/kotlin/soda/kotlin/web/MessageWriter.kt
package soda.kotlin.web

import com.sun.net.httpserver.HttpExchange
import java.nio.charset.StandardCharsets

interface MessageWriter {

    fun sendMessage(exchange: HttpExchange, code: Int, message: String) {
        val data = message.toByteArray(StandardCharsets.UTF_8)
        exchange.sendResponseHeaders(code, data.size.toLong())
        exchange.responseBody.write(data)
        exchange.responseBody.close()
    }

    fun sendMessageWithCatch(exchange: HttpExchange, code: Int, message: String) {
        try {
            sendMessage(exchange, code, message)
        } catch (ex: Throwable) {
            Logger.exception("message sending error, code - $code, msg - $message", ex)
        }
    }

}